tools/ReleaseProcess/UploadArtifactsToGitHub.ps1

```diff
@@ -0,0 +1,444 @@
+            continue;
+        }
+
+        # Renaming
+        foreach ($package in $packages)
+        {
+            echo "Processing : $($package.FullName)"
+            # Deconstruct the artifact name
+            $name,$version,$arch,$suffix = $package.Name.split("_")
+            $arch,$ext,$suffix = $arch.split(".")
+
+            # Reconstruct the new name from the segments.
+            $finalName = @($name, $version, $os, $arch) -join '_'
+            $finalName = "$finalName.$ext"
+            $newPath = $(Join-Path -Path "$($package.Directory)" -ChildPath "$finalName")
+
+            # Rename
+            Rename-Item -Path $package.FullName -NewName $newPath
+
+            # Record renamed files
+            $artifactFinalists += $(Get-Item $newPath)
+        }
+
+        echo ""
+    }
+
+    echo ""
+
+    # Stage uploading files
+    foreach ($artifact in $artifactFinalists)
+    {
+        echo "Moving : $($artifact.FullName)"
+        echo "To : $(Join-Path -Path $outputDir -ChildPath $artifact.Name)"
+        Move-Item -Path $artifact.FullName -Destination $(Join-Path -Path $outputDir -ChildPath $artifact.Name) -Force
+    }
+}
+
+
+<#
+    .SYNOPSIS
+        This function downloads and prepares the IoT Identity Service artifacts from Github Action for a release.
+        The final artifacts are collected under "<WorkDir>/output" path.
+    .NOTES
+        Requires Azure CLI to be login
+    .DESCRIPTION
+        1) Download edgelet artifacts from a given commit off of the Github build action:
+              https://github.com/Azure/iot-identity-service/actions/workflows/packages.yaml
+        2) Extract the *.zip artifacts into their own directories
+        3) Traverse the directories and rename a relavant artifact into the following format: 
+              .DEB _________________ {component}_{version}_{os}_{arch}.deb 
+              .RPM _________________ {component}-{version}.{os}.{arch}.rpm  (No name change)
+        4) Move the renamed artifacts to a staging directory at "<WorkDir>/output"
+    .PARAMETER CommitId
+        Github SHA256 commit id
+    .PARAMETER WorkDir
+        Absolute path of the working directory
+#>
+function Upload-Artifacts-To-GitHub
+{
+    [CmdletBinding()]
+    param (
+        <# 
+        Absolute path of current working directory
+        #>
+        [Parameter(Mandatory)]
+        [string]
+        $WorkDir,
+
+        <# 
+        Branch name of "iotedge" repository.
+        This parameter is used for the script to figure out the releasing version.
+        i.e. "master", "release/1.1", "release/1.2", etc.
+        #>
+        [Parameter(Mandatory)]
+        [string]
+        $BranchName
+    )
+
+    $pat = $(az keyvault secret show -n GitHubAccessToken --vault-name edgebuildkv | ConvertFrom-Json)
+    $pat = $pat.value;
+
+    # Get the latest release from a given branch
+    echo "Fetch the latest release: "
+    $url = "https://api.github.com/repos/Azure/iotedge/releases"
```

Comment:
I think it would be more correct to fetch the latest release from the Azure/azure-iotedge repo instead of Azure/iotedge.

Reply:
For the release in `Azure/azure-iotedge` every release is done against `master` branch. I cannot systematically query the latest release from a given branch there.